src/Core/src/Handlers/DatePicker/DatePickerHandler.Android.cs

```diff
@@ -142,10 +142,16 @@ void ShowPickerDialog(int year, int month, int day)
 				_dialog.UpdateDate(year, month, day);
 
 			_dialog.CancelEvent += OnCancelButtonClicked;
+			_dialog.DismissEvent += OnDialogDismissed;
 
 			_dialog.Show();
 		}
 
+		private void OnDialogDismissed(object? sender, EventArgs e)
+		{
+			_dialog = null;
```

Comment:
If we don't find another way to fix this and _do_ have to re-create the dialog every time we display it, we'll need to update this PR to account for `_dialog` possible being `null` when `MapMinimumDate` or `MapMaximumDate` are called.

Reply:
This is no longer relevant in the new solution.